app/build.gradle.kts

```diff
@@ -75,7 +80,9 @@ android {
         release {
             isMinifyEnabled = false
 
-            signingConfig = signingConfigs.getByName("release")
+            if (useKeystoreProperties) {
```

Comment:
Is this change really necessary? For now, we just want to build as debug.

Reply:
This is nessesary, because:

1. Otherwise the build will fail
2. We can obtain signed debug and unsigned release (in artifacts) which can be signed by whoever wants with apktool

Alternatively, we can skip this and then build with command `./gradlew build -x assembleRelease --no-daemon` on ci, though I prefer to have both debug and release, even if it's unsigned.

Reply:
Understood. But why do we want to generate a release APK in a pull request to develop? I think the correct time to generate it would be at the creation of a release, when merging into master

Reply:
If you really don't want to upload release apk as an artifact, we can change it in the workflow. But this `is` instruction is needed anyway because the project is not building properly if keystore.properties is absent. Let users build whatever they want either signed or unsigned.

Reply:
Isn't it enough to run a buildDebug? Or assembleDebug, since we don't have unit tests yet.

Reply:
Ok, we can run just `assembleDebug`, although it will skip lint.

Alternatively, we can `./gradlew check assembleDebug --no-daemon`.

Reply:
I haven't set up any lint in the project yet

Reply:
You don't have to, AGP does it in basic way. For example, if you add some strings in `values/strings.xml` and don't translate it, it throws an error.

Lint reports are at `app/build/reports/lint-results-debug.html`; at the current stage of development it doessn't show any error. Now two wornings from dev branch shows obsolete dependency which are updated in #116.

<https://proandroiddev.com/effectively-using-android-lint-in-existing-projects-eacff100a919>

To sum up, lint (`./gradlew check` or `./gradlew lint` is capable of identifying most of trivial errors such as untranslated strings, some deprecated APIs, images without alternative text etc. Further more, if lint throws an error and `lint-baseline.xml` is not present, or other annotations in the code, the `./gradlew build` will fail and won't create apk. That's `lint.abortOnError = false` exists in some projects build.gradle.kts files.

Reply:
Man, that's awesome. I'm learning a lot from you.

Reply:
Well, we can manually run the steps that assembleDebug skips, right?

Reply:
Well, I leave the decision up to you. But if you decide to keep the check, I suggest the following:

create this extension in buildSrc
```kotlin
fun Project.hasFile(name: String): Boolean {
    return rootProject.file(name).canRead()
}
```

and check it in the following way:
```kotlin
if (hasFile(name = "keystore.properties")) {
    signingConfig = signingConfigs.getByName("release")
}
```

This will make the build code less cluttered, as most projects are so messy in this part.

Reply:
I'm learning a lot from you, too. Did as you suggested in recent commits.